smartapps/smartthings/virtual-device-manager.src/virtual-device-manager.groovy

```diff
@@ -0,0 +1,78 @@
+/**
+ *  Virtual Device Manager
+ *
+ *  Copyright 2015 Bob Florian/SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+definition(
+    name: "Virtual Device Manager",
+    namespace: "smartthings",
+    author: "SmartThings",
+    description: "Tests child device creation and update",
+    iconUrl: "https://s3.amazonaws.com/smartapp-icons/Convenience/Cat-Convenience.png",
+    iconX2Url: "https://s3.amazonaws.com/smartapp-icons/Convenience/Cat-Convenience@2x.png",
+    iconX3Url: "https://s3.amazonaws.com/smartapp-icons/Convenience/Cat-Convenience@2x.png"
+)
+
+
+preferences {
+    page name: "mainPage", install: true, uninstall: true
+
+}
+
+def mainPage() {
+    dynamicPage(name: "mainPage") {
+        section {
+            input "switchCount", "number", title: "Select number of Virtual Switches", defaultValue: 1, required: true
+            input "dimmerCount", "number", title: "Select number of Virtual Dimmers", defaultValue: 1, required: true
+            input "theHub", "hub", title: "Select hub for local execution", multiple: false, required: false
+        }
+    }
+}
+
+def installed() {
+    log.debug "Installed with settings: ${settings}"
+    state.lastSwitchDni = 0
+    state.lastDimmerDni = 0
```

Comment:
I think we should use a single counter and name for these devices (ex: virtual-$num) because they may get retyped in the future which will lead to confusion if the DNI doesn't match the type.